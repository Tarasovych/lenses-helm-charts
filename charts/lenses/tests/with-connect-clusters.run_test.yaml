suite: enable connect clusters and check the changes in secret
templates:
  - secrets.yaml
tests:
  - it: should work
    values:
      - values/with-connect-clusters.yaml
    asserts:
      # Check the rendered manifest has not changed
      # Review the changes and if they are verified use `helm unittest -u .` to update the manifest
      # Writing tests that cover this chaneg is highly recommended
      - matchSnapshot: {}
  - it: should work with explicit Connect metrics URL
    values:
      - values/with-connect-metrics-url.yaml
    asserts:
      # Check the rendered manifest has not changed
      # Review the changes and if they are verified use `helm unittest -u .` to update the manifest
      # Writing tests that cover this chaneg is highly recommended
      - matchSnapshot: {}
      - equal:
          path: stringData.connect\.clusters
          value: |-
            [
              {
                name: "streamreactor",
                statuses: "connect-status",
                configs: "connect-configs",
                offsets: "connect-offsets",
                aes256: { key: "PasswordPasswordPasswordPassword" },
                urls: [
                  {
                    url: "http://connect-worker-1:8083",
                    metrics: {
                      url: "http://connect-worker-1:8080/metrics",
                      type: "JMX",
                      ssl: false,
                      user: "admin",
                      password: "admin"
                    }
                  }
                ]
              }
            ]


