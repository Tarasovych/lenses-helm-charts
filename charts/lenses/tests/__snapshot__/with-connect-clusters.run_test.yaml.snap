should work:
  1: |
    apiVersion: v1
    data:
      client.keystore.jks: ""
      client.truststore.jks: ""
      jvm.truststore.jks: ""
      keytab: ""
      lenses.keytab: ""
      lenses.opts.keystore.jks: ""
      lenses.opts.truststore.jks: ""
      lenses.tls.keystore.jks: ""
      lenses.tls.truststore.jks: ""
      saml.idp.xml: ""
      saml.keystore.jks: ""
      schema.registry.keystore.jks: ""
      schema.registry.truststore.jks: ""
    kind: Secret
    metadata:
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.18
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-secret
        release: RELEASE-NAME
      name: RELEASE-NAME
    stringData:
      client.key.password: ""
      client.keystore.password: ""
      client.truststore.password: ""
      connect.clusters: |-
        [
          {
            name: "streamreactor",
            statuses: "connect-status",
            configs: "connect-configs",
            offsets: "connect-offsets",
            aes256: { key: "PasswordPasswordPasswordPassword" },
            urls: [
              {
                url: "http://connect-worker-1:8083",
                metrics: {
                  url: "connect-worker-1:9102",
                  type: "JMX",
                  ssl: false,
                  user: "admin",
                  password: "admin"
                }
              },
              {
                url: "http://connect-worker-2:8083"
              },
              {
                url: "http://connect-worker-3:8083"
              }
            ]
          },
          {
            name: "sqllenses",
            statuses: "connect-sqllenses-status",
            configs: "connect-sqllenses-configs",
            offsets: "connect-sqllenses-offsets",
            urls: [
              {
                url: "http://sqllenses-worker-1:8084"
              },
              {
                url: "http://sqllenses-worker-2:8084"
              }
            ]
          }
        ]
      jaas.conf: ""
      lenses.opts.keystore.password: ""
      lenses.opts.truststore.password: ""
      lenses.tls.key.password: ""
      lenses.tls.keystore.password: ""
      lenses.tls.truststore.password: ""
      license: ""
      schema.registry.key.password: ""
      schema.registry.keystore.password: ""
      schema.registry.truststore.password: ""
      security.append.conf: ""
      security.conf: ""
should work with explicit Connect metrics URL:
  1: |
    apiVersion: v1
    data:
      client.keystore.jks: ""
      client.truststore.jks: ""
      jvm.truststore.jks: ""
      keytab: ""
      lenses.keytab: ""
      lenses.opts.keystore.jks: ""
      lenses.opts.truststore.jks: ""
      lenses.tls.keystore.jks: ""
      lenses.tls.truststore.jks: ""
      saml.idp.xml: ""
      saml.keystore.jks: ""
      schema.registry.keystore.jks: ""
      schema.registry.truststore.jks: ""
    kind: Secret
    metadata:
      labels:
        app: RELEASE-NAME
        chart: lenses-4.2.18
        heritage: Helm
        lenses.io/app: RELEASE-NAME
        lenses.io/app.type: lenses-secret
        release: RELEASE-NAME
      name: RELEASE-NAME
    stringData:
      client.key.password: ""
      client.keystore.password: ""
      client.truststore.password: ""
      connect.clusters: |-
        [
          {
            name: "streamreactor",
            statuses: "connect-status",
            configs: "connect-configs",
            offsets: "connect-offsets",
            aes256: { key: "PasswordPasswordPasswordPassword" },
            urls: [
              {
                url: "http://connect-worker-1:8083",
                metrics: {
                  url: "http://connect-worker-1:8080/metrics",
                  type: "JMX",
                  ssl: false,
                  user: "admin",
                  password: "admin"
                }
              },
              {
                url: "http://connect-worker-2:8083/custom/path"
              }
            ]
          }
        ]
      jaas.conf: ""
      lenses.opts.keystore.password: ""
      lenses.opts.truststore.password: ""
      lenses.tls.key.password: ""
      lenses.tls.keystore.password: ""
      lenses.tls.truststore.password: ""
      license: ""
      schema.registry.key.password: ""
      schema.registry.keystore.password: ""
      schema.registry.truststore.password: ""
      security.append.conf: ""
      security.conf: ""
